src/java/org/apache/cassandra/index/IndexRegistry.java

```diff
@@ -356,66 +383,79 @@ enum EqBehavior
     class EqBehaviorIndexes
     {
         public EqBehavior behavior;
-        public final Index eqIndex;
-        public final Index matchIndex;
+        public final Collection<Index> eqIndexes;
+        public final Collection<Index> matchIndexes;
 
-        private EqBehaviorIndexes(Index eqIndex, Index matchIndex, EqBehavior behavior)
+        private EqBehaviorIndexes(Collection<Index> eqIndexes, Collection<Index> matchIndexes, EqBehavior behavior)
         {
-            this.eqIndex = eqIndex;
-            this.matchIndex = matchIndex;
+            this.eqIndexes = eqIndexes;
+            this.matchIndexes = matchIndexes;
             this.behavior = behavior;
         }
 
-        public static EqBehaviorIndexes eq(Index eqIndex)
+        public static EqBehaviorIndexes eq(Collection<Index> eqIndexes)
         {
-            return new EqBehaviorIndexes(eqIndex, null, EqBehavior.EQ);
+            return new EqBehaviorIndexes(eqIndexes, null, EqBehavior.EQ);
         }
 
-        public static EqBehaviorIndexes match(Index eqAndMatchIndex)
+        public static EqBehaviorIndexes match(Collection<Index> eqAndMatchIndexes)
         {
-            return new EqBehaviorIndexes(eqAndMatchIndex, eqAndMatchIndex, EqBehavior.MATCH);
+            return new EqBehaviorIndexes(eqAndMatchIndexes, eqAndMatchIndexes, EqBehavior.MATCH);
         }
 
-        public static EqBehaviorIndexes ambiguous(Index firstEqIndex, Index secondEqIndex)
+        public static EqBehaviorIndexes ambiguous(Collection<Index> firstEqIndexes, Collection<Index> secondEqIndexes)
         {
-            return new EqBehaviorIndexes(firstEqIndex, secondEqIndex, EqBehavior.AMBIGUOUS);
+            return new EqBehaviorIndexes(firstEqIndexes, secondEqIndexes, EqBehavior.AMBIGUOUS);
         }
     }
 
     /**
      * @return
-     * - AMBIGUOUS if an index supports EQ and a different one supports both EQ and ANALYZER_MATCHES
-     * - MATCHES if an index supports both EQ and ANALYZER_MATCHES
+     * - AMBIGUOUS if an index that is not excluded by the hints supports EQ and a different one, also not exluded by
+     *   the hints, supports both EQ and ANALYZER_MATCHES. If any of the indexes is included by the hints, the behavior
```

Comment:
```
If any of the indexes is included by the hints
```
sounds that if both indexes are included then it's not ambiguous. If I don't mistake, only one should be hinted.